{
  if (LOG.isDebugEnabled()) {
    LOG.debug("Running " + GenericTestUtils.getMethodName());
  }
  DataNode spyDN=spy(dn);
  doThrow(new IOException()).when(spyDN).updateReplicaUnderRecovery(block,RECOVERY_ID,BLOCK_ID,block.getNumBytes());
  try {
    spyDN.syncBlock(rBlock,initBlockRecords(spyDN));
    fail("Sync should fail");
  }
 catch (  IOException e) {
    e.getMessage().startsWith("Cannot recover ");
  }
}
